contracts/IntentSource.sol

```diff
@@ -313,52 +315,92 @@ contract IntentSource is IIntentSource, Semver {
         }
     }
 
+    /**
+     * @notice Refunds rewards to the intent creator
+     * @param routeHash Hash of the intent's route
+     * @param reward Reward structure of the intent
+     */
+    function refundIntent(bytes32 routeHash, Reward calldata reward) external {
+        bytes32 rewardHash = keccak256(abi.encode(reward));
+        bytes32 intentHash = keccak256(abi.encodePacked(routeHash, rewardHash));
+
+        VaultState memory state = vaults[intentHash].state;
+
+        if (
+            state.status != uint8(RewardStatus.Claimed) &&
+            state.status != uint8(RewardStatus.Refunded) &&
+            block.timestamp <= reward.deadline
+        ) {
+            revert IntentNotExpired(intentHash);
+        }
+
+        if (state.status != uint8(RewardStatus.Claimed)) {
+            state.status = uint8(RewardStatus.Refunded);
+        }
+
+        state.mode = uint8(VaultMode.Refund);
+        state.allowPartialFunding = 0;
+        state.isPermit2 = 0;
+        state.target = address(0);
+        vaults[intentHash].state = state;
+
+        emit Refund(intentHash, reward.creator);
+
+        new IntentVault{salt: routeHash}(intentHash, reward);
+    }
+
     /**
      * @notice Refunds rewards to the intent creator
      * @param routeHash Hash of the intent's route
      * @param reward Reward structure of the intent
      * @param token Optional token address for handling incorrect vault transfers
      */
-    function refundIntent(
+    function refundToken(
```

Comment:
We may need to rename this function to `recoverToken` and update the SolDoc. I believe it is used to recover funds that are not part of the intent. 

Since this is a recovery function, the `Refund` event shouldn't emitted here.